src/plugins/chart_expressions/expression_heatmap/public/compoments/heatmap_component.test.tsx

```diff
@@ -0,0 +1,247 @@
+      range: 'number',
+    },
+  },
+  showTooltip: true,
+  highlightInHover: false,
+  xAccessor: 'col-1-2',
+  valueAccessor: 'col-0-1',
+};
+const data: Datatable = {
+  type: 'datatable',
+  rows: [
+    { 'col-0-1': 0, 'col-1-2': 'a' },
+    { 'col-0-1': 148, 'col-1-2': 'b' },
+  ],
+  columns: [
+    { id: 'col-0-1', name: 'Count', meta: { type: 'number' } },
+    { id: 'col-1-2', name: 'Dest', meta: { type: 'string' } },
+  ],
+};
+
+const mockState = new Map();
+const uiState = {
+  get: jest
+    .fn()
+    .mockImplementation((key, fallback) => (mockState.has(key) ? mockState.get(key) : fallback)),
+  set: jest.fn().mockImplementation((key, value) => mockState.set(key, value)),
+  emit: jest.fn(),
+  setSilent: jest.fn(),
+} as any;
+
+describe('PieComponent', function () {
+  let wrapperProps: HeatmapRenderProps;
+
+  beforeAll(() => {
+    wrapperProps = {
+      data,
+      chartsThemeService,
+      args,
+      uiState,
+      onClickValue: jest.fn(),
+      onSelectRange: jest.fn(),
+      paletteService: palettesRegistry,
+      formatFactory: formatService.deserialize,
+    };
+  });
+
+  it('renders the legend on the correct position', () => {
+    const component = shallowWithIntl(<HeatmapComponent {...wrapperProps} />);
+    expect(component.find(Settings).prop('legendPosition')).toEqual('top');
+  });
+
+  it('renders the legend toggle component if uiState is set', async () => {
+    const component = mountWithIntl(<HeatmapComponent {...wrapperProps} />);
+    await act(async () => {
+      expect(findTestSubject(component, 'vislibToggleLegend').length).toBe(1);
+    });
+  });
+
+  it('not renders the legend toggle component if uiState is undefined', async () => {
+    const newProps = { ...wrapperProps, uiState: undefined } as unknown as HeatmapRenderProps;
+    const component = mountWithIntl(<HeatmapComponent {...newProps} />);
+    await act(async () => {
+      expect(findTestSubject(component, 'vislibToggleLegend').length).toBe(0);
+    });
+  });
+
+  it('renders the legendColorPicker if uiState is set', async () => {
+    const component = mountWithIntl(<HeatmapComponent {...wrapperProps} />);
+    await act(async () => {
+      expect(component.find(Settings).prop('legendColorPicker')).toBeDefined();
+    });
+  });
+
+  it('not renders the legendColorPicker if uiState is undefined', async () => {
+    const newProps = { ...wrapperProps, uiState: undefined } as unknown as HeatmapRenderProps;
+    const component = mountWithIntl(<HeatmapComponent {...newProps} />);
+    await act(async () => {
+      expect(component.find(Settings).prop('legendColorPicker')).toBeUndefined();
+    });
+  });
+
+  it('computes the bands correctly for infinite bounds', async () => {
+    const component = mountWithIntl(<HeatmapComponent {...wrapperProps} />);
+    await act(async () => {
+      expect(component.find(Heatmap).prop('colorScale')).toEqual({
+        bands: [
+          { color: 'rgb(0, 0, 0, 0)', end: 0, start: 0 },
```

Comment:
Isn't this `rgb` string wrong? It think it should be  either`rgb(0,0,0)` or `rgba(0,0,0,0)`